ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerFragment.kt

```diff
@@ -128,6 +130,29 @@ class HomeScreenMapContainerFragment : AbstractMapContainerFragment() {
     super.onViewCreated(view, savedInstanceState)
     setupMenuFab()
     setupBottomLoiCards()
+    lifecycleScope.launch {
+      mapContainerViewModel.activeSurvey
+        .combine(mapContainerViewModel.lois) { x, y -> Pair(x, y) }
+        .collect {
+          val survey = it.first
+          val lois = it.second
+          survey?.let {
+            val suggest =
+              it.jobs.any {
+                it.strategy != Job.DataCollectionStrategy.PREDEFINED &&
+                  it.strategy != Job.DataCollectionStrategy.UNKNOWN
+              }
+            val readonly = lois.isEmpty()
+            val msg =
+              when {
+                suggest -> resources.getString(R.string.suggest_data_collection_hint)
```

Comment:
I find the indirection in these vars and conditions hard to grok and keep in my head all at once. Instead, perhaps we can write:

```kt
EphemeralPopups.dismissableSnackbar(binding.root, zoomHintId(survey, lois))
...
fun zoomHintId(survey, lois) {
val msgId = when {
  survey.jobs.any { it.canDataCollectorsAddLois() } -> R.string.ad_hoc_data_collection zoom_hint
  lois.isEmpty() -> R.string.read_only_data_collection_zoom_hint
  else -> R.string.predefined_data_collection_zoom_hint
}

}
```